### ISSUE TITLE
Cannot include pdf in any possible way

### ISSUE THREAD

**Issue by Dzordzu** [tectonic-typesetting/tectonic#1085]
## Details
Tectonic: 0.14.1

## Example

* mydoc exists within the same directory
* tried also with relative paths

```latex
\includepdf{mydoc.pdf}
```

OR

```latex
\includegraphics{mydoc.pdf}
```

Both fail with
```
warning: ICC profile format spec. version 4.4.0 not supported in current PDF version setting.
warning: ICC profile not embedded.
warning: Didn't find "endobj".
error: pdf_link_obj(): passed invalid object.
```

## Fix

A PDF was broken (missing `endobj`). In order to fix this I used [ILovePDF](https://www.ilovepdf.com/). However, I would like to keep this issue open until even "broken" PDFs are included (pdflatex, latexmk and many pdf viewers were working with the faulted PDF). Obviously, in case of maintainers disapproval towards an idea of including "hacks" that would allow to include faulty documents I'll instantly close this issue 

For onlookers: In case of broken pdf, you may also want to use opensource tools and methods described within [superuser thread](https://superuser.com/questions/278562/how-can-i-fix-repair-a-corrupted-pdf-file)

## Keywords

* Broken PDF
* Include issue
* PDF insertion error
* PDF import error
* endobj 

## References

* #854
* #50

**Comment by omnv** [comment]
Hey @Dzordzu thx for going through other issues before opening this one!

So far I'm not able to reproduce it with a _dummy PDF_:
```latex
\documentclass{article}
  \usepackage{graphicx}

\begin{document}
  Included PDF:
  \includegraphics{to-include.pdf}
\end{document}
```
Can you try including a different PDF generated by LaTeX?

**Comment by Dzordzu** [comment]
Okey. Strange. The tex based pdf was easily included… But this still does not resolve my issue :/

**Comment by Dzordzu** [comment]
What's strange, it's also *does* work with my awesome, example PDF [smiley.pdf](https://github.com/tectonic-typesetting/tectonic/files/12549933/smiley.pdf)

**Comment by Dzordzu** [comment]
Okey. It seems that my pdf is somehow missing `endobj`. The strange thing is that pdflatex and other tools were generating document properly.

**Comment by omnv** [comment]
> > Can you try including a different PDF generated by LaTeX?
> 
> Is it a required constraint? I've got an external document that was certainly NOT generated by LaTeX.

Not a required constraint! Just wanted to narrow down the issue.

**Comment by omnv** [comment]
@Dzordzu Is Interesting! was `xelatex` (`latexmk -pdfxe test.tex`) also working properly with your external document?

**Comment by Dzordzu** [comment]
@omnv Yep. It was working fine, but was also generating (as any other tool) lots of errors and warnings. Well certainly, one of them was about missing `endobj` ;) I got the working PDF with the file included

@vlasakm It's the scan of my [old recommendation letter](https://github.com/tectonic-typesetting/tectonic/files/12551503/recommendation.pdf). I tried to manually find missing `endobj`, but I was unsuccessful
